utils/staking-miner/src/main.rs

```diff
@@ -420,17 +421,37 @@ fn mine_dpos<T: EPM::Config>(ext: &mut Ext) -> Result<(), Error<T>> {
 
 pub(crate) async fn check_versions<T: frame_system::Config + EPM::Config>(
 	rpc: &SharedRpcClient,
+	print: bool,
 ) -> Result<(), Error<T>> {
 	let linked_version = T::Version::get();
 	let on_chain_version = rpc
 		.runtime_version(None)
 		.await
 		.expect("runtime version RPC should always work; qed");
 
-	log::debug!(target: LOG_TARGET, "linked version {:?}", linked_version);
-	log::debug!(target: LOG_TARGET, "on-chain version {:?}", on_chain_version);
+	let do_print = || {
+		log::info!(
+			target: LOG_TARGET,
+			"linked version {:?}",
+			(&linked_version.spec_name, &linked_version.spec_version)
+		);
+		log::info!(
+			target: LOG_TARGET,
+			"on-chain version {:?}",
+			(&on_chain_version.spec_name, &on_chain_version.spec_version)
+		);
+	};
+
+	if print {
+		do_print();
+	}
 
-	if linked_version != on_chain_version {
+	// we relax the checking here a bit, which should not cause any issues in production (a chain
+	// that messes up its spec name is highly unlikely), but it allows us to do easier testing.
```

Comment:
About messed up spec name; I accidentally ran the try-runtime with the wrong `--chain` argument.  
So I would run Kusama migrations on a Polkadot node and it worked :see_no_evil:   
Could we maybe add an extra flag (not in this MR) that controls whether a different spec name should panic?

Reply:
we already have something very similar: https://paritytech.github.io/substrate/master/try_runtime_cli/struct.SharedParams.html#